issuer-web/src/store/modules/connection/actions.ts

```diff
@@ -56,5 +56,45 @@ export const actions: ActionTree<ConnectionState, RootState> = {
           reject(error);
         });
     });
+  },
+  async waitForConnectionStatus(
+    context: ActionContext<ConnectionState, RootState>,
+    status: ConnectionStatus
+  ): Promise<ConnectionStatus> {
+    const config: AppConfig = await ConfigService.getAppConfig();
+    return new Promise<ConnectionStatus>((resolve, reject) => {
+      const id = context.getters["getConnection"].id;
+      const retryInterceptor = Axios.interceptors.response.use(
+        response => {
+          const responseData = response.data as AgentConnectionStatusInterface;
+          if (responseData.state === status) {
+            Axios.interceptors.request.eject(retryInterceptor);
+            return response;
+          } else {
+            // retry until the credential has not been issued
+            return Axios.request(response.config);
+          }
+        },
+        error => {
+          // Any status codes that falls outside the range of 2xx cause this function to trigger
+          console.log("An error occurred while fetching the connection status");
+          return Promise.reject(error);
+        }
+      );
+      Axios.get(`${config.apiServer.url}/connections/${id}`)
+        .then(response => {
+          if (response.status === 200) {
+            const responseData = response.data as AgentConnectionStatusInterface;
+            context.commit("setStatus", responseData.state);
+            resolve(responseData.state);
+          } else {
+            throw "An error occurred while fetching the connection status";
```

Comment:
I don't think throwing string is kosher anymore.. can't it just reject the promise at that point?

Reply:
Shouldn't this result in the catch block (that follows) picking up the error and rejecting with this message?
I can definitely reject here as well, but was trying to keep the exit points for the function limited (e.g.: always exit on the catch block for errors). Thoughts?

Reply:
Yes, it will do that. My eslint rules disallow throwing a string though :)
Throwing a new Error(msg) gives you a stack trace, which is helpful when the source of the error isn't known.
Only having one reject is probably a good idea, you're right.

Reply:
I'll switch to throwing an actual error, tweaking another couple of things and I'll push soon :)